src/compiler/scala/tools/nsc/typechecker/Typers.scala

```diff
@@ -4160,7 +4156,7 @@ trait Typers extends Adaptations with Tags with TypersTracking with PatternTyper
           val info = annInfo(typedAnn)
           // check message and since args to deprecated
           // when deprecated gets more params, those other args may also be defaults.
-          def oneTwoPunch(p: Tree => Boolean) = (0 to 1).exists(info.argAtIndex(_).map(p).getOrElse(false))
+          def oneTwoPunch(p: Tree => Boolean) = info.argAtIndex(0).exists(p) || info.argAtIndex(1).exists(p)
```

Comment:
It slightly dilutes the pun. :) The previous comment was expecting they'd add `forRemoval` to deprecated; it wasn't necessary to hardcode the expected count. Also FTR the `blockUsesDefault` test that follows is hard to follow.

Reply:
You mean it dilutes the punch? 😄